python/tvm/relay/frontend/onnx.py

```diff
@@ -410,8 +410,16 @@ class Pool(OnnxOpConverter):
 
     @classmethod
     def _impl_v1(cls, inputs, attr, params):
+        data = inputs[0]
+        input_shape = infer_shape(data)
+        ndim = len(input_shape)
+
         attr_cvt, data = cls._run_calculation(inputs, attr, params)
-        return attr_cvt([data], attr, params)
+        out = attr_cvt([data], attr, params)
+
+        if ndim == 3 and len(attr["kernel_shape"]) == 2:
```

Comment:
Can you make this so if kernel_shape rank and ndim differ by 1 we assume it's missing a batch dimension and runs below code? (same thing for _run_calculation